src/malli/core.cljc

```diff
@@ -433,50 +430,43 @@
 ;; Schemas
 ;;
 
-(defn -simple-schema [?props]
-  (let [props* (atom (if (map? ?props) ?props))]
-    ^{:type ::into-schema}
-    (reify IntoSchema
-      (-type [_] (:type @props*))
-      (-type-properties [_] (:type-properties @props*))
-      (-properties-schema [_ _])
-      (-children-schema [_ _])
-      (-into-schema [parent properties children options]
-        (if (fn? ?props)
-          (-into-schema (-simple-schema (?props properties children)) properties children options)
-          (let [type (if ?props (?props :type))
-                pred (if ?props (?props :pred))
-                property-pred (if ?props (?props :property-pred))
-                min (if ?props (?props :min 0) 0)
-                max (if ?props (?props :max 0) 0)]
-            (reset! props* ?props)
-            (-check-children! type properties children {:min min, :max max})
-            (let [pvalidator (if property-pred (property-pred properties))
-                  validator (if pvalidator (fn [x] (and (pred x) (pvalidator x))) pred)
-                  form (-create-form type properties children)]
-              ^{:type ::schema}
-              (reify
-                Schema
-                (-validator [_] validator)
-                (-explainer [this path]
-                  (fn explain [x in acc]
-                    (if-not (validator x) (conj acc (miu/-error path in this x)) acc)))
-                (-parser [_] (fn [x] (if (validator x) x ::invalid)))
-                (-unparser [this] (-parser this))
-                (-transformer [this transformer method options]
-                  (-intercepting (-value-transformer transformer this method options)))
-                (-walk [this walker path options]
-                  (if (-accept walker this path options)
-                    (-outer walker this path children options)))
-                (-properties [_] properties)
-                (-options [_] options)
-                (-children [_] children)
-                (-parent [_] parent)
-                (-form [_] form)
-                LensSchema
-                (-keep [_])
-                (-get [_ _ default] default)
-                (-set [this key _] (-fail! ::non-associative-schema {:schema this, :key key}))))))))))
+(defn -simple-schema [{:keys [type type-properties pred property-pred min max] :or {min 0, max 0} :as ?props}]
```

Comment:
@ikitommi  What about the case where it's used with a function constructor?
I know it's probably a noop but isn't it confusing?

Reply:
could be `?props` and destructured in a let?

Reply:
good point, will change that.

Reply:
fixed in 91773e9